app/controllers/api/v1/study_files_controller.rb

```diff
@@ -478,7 +479,9 @@ def parse
               @study_file.update(parse_status: 'parsing')
               genes.update(parse_status: 'parsing')
               barcodes.update(parse_status: 'parsing')
-              ParseUtils.delay.cell_ranger_expression_parse(@study, current_api_user, @study_file, genes, barcodes)
+              @study_file.update(parse_status: 'parsing')
```

Comment:
This duplicates line 479.

```suggestion
```

More generally, `@study_file.update(parse_status: 'parsing')` is called every time before `IngestJob.new`, and `job.delay.push_remote_and_launch_ingest` is called every time after.


To eliminate duplicate code, can those two calls be moved within the `IngestJob` constructor?

Reply:
I agree about the `parse_status` update, but I don't actually want `push_remote_and_launch_ingest` to be invoked when we call `IngestJob.new`.  There is an undocumented but highly useful side effect of being able to instantiate that class without invoking ingest that allows us to spot-check running jobs easily, or even retrieve PAPI-level error messages from failed runs (even days or weeks later, as the job doesn't get purged for quite a while).  If it invoked an ingest run every time `initialize` gets called, that would cause both to runs to fail, or uselessly retry a previously failed run.

Reply:
I've created SCP-2659 to track this and some related ideas directly - there's a great deal of duplicated code surrounding parsing logic that needs to be refactored that I think is a PR in its own right.